### PR TITLE
fix: keep accordion open on price refresh

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- It was reported that when the accordion was open and the price refreshed after user's input, it would close the accordion. This has been fixed:

https://github.com/user-attachments/assets/b919bd9c-6118-4905-ab9e-4dfaa8ad4c60

- Should work for SWAP and LIMIT
- When changing a token (route), it is currently expected that the accordion is closed again.

### CODE REVIEW

File: apps/cowswap-frontend/src/common/pure/TradeDetailsAccordion/index.tsx [hunk 1]
```diff
@@ -1,135 +1,57 @@
-import { useState } from 'react'
+import React from 'react'
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/TradeDetailsAccordion/index.tsx:1]:
```suggestion
```

File: apps/cowswap-frontend/src/common/pure/TradeDetailsAccordion/index.tsx [hunk 2]
```diff
@@ -1,135 +1,57 @@
...
-  const [isOpen, setIsOpen] = useState(open)
-
-  const toggleAccordion = () => {
-    setIsOpen(!isOpen)
+  const handleToggle = () => {
+    if (typeof onToggle === 'function') {
+      onToggle()
+    } else {
+      console.error('onToggle is not a function:', onToggle)
+    }
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/TradeDetailsAccordion/index.tsx:33]: Why do you expect it not to be a function?
According to the type, it should always be one.

You could also:

```suggestion
    onToggle?.()
```

Comment (fairlighteth) [reply apps/cowswap-frontend/src/common/pure/TradeDetailsAccordion/index.tsx:33]: Thanks for the suggestion! I think this was a debugging feature I didn't pay attention to anymore.

**Comment by fairlighteth** [comment]
> Hey @fairlighteth , changes LGTM! However, 1 question: when I go to the Confirm screen and return back, I see accordion collapsed. Is this an expected behavior?

Yes, in line with when you switch tokens, in the current implementation it will close the fee accordion. In the future might want to iterate on this logic further.

**Review by alfetopito — APPROVED** [review]

**Pull request merged**